crate/server/src/tests/google_cse/mod.rs

```diff
@@ -0,0 +1,292 @@
+    ))
+    .unwrap();
+
+    let rsa_private_key = Rsa::<Private>::private_key_from_pem(&blue_private_key)?;
+    let private_key = PKey::from_rsa(rsa_private_key)?;
+    let mut signer = Signer::new(MessageDigest::sha256(), &private_key)?;
+
+    tracing::debug!("padding method: {:?}", signer.rsa_padding());
+
+    signer.update(&digest)?;
+    let signature = signer.sign_to_vec()?;
+
+    tracing::debug!(
+        "signature: {}",
+        general_purpose::STANDARD.encode(signature.clone())
+    );
+
+    //-------------------------------------------------------------------------
+    // Verify
+    //-------------------------------------------------------------------------
+    // The RSA blue public key
+    let blue_public_key = read_bytes_from_file(&PathBuf::from(
+        "src/routes/google_cse/python/openssl/blue.pem",
+    ))
+    .unwrap();
+    let rsa_public_key = X509::from_pem(&blue_public_key)?;
+    let public_key = rsa_public_key.public_key()?;
+    // Verify the signature
+    let mut verifier = Verifier::new(MessageDigest::sha256(), &public_key)?;
+    verifier.update(&digest)?;
+
+    assert!(verifier.verify(&signature)?);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_cse_private_key_sign() -> KResult<()> {
+    std::env::set_var("KMS_GOOGLE_CSE_GMAIL_JWKS_URI", JWKS_URI);
+    std::env::set_var("KMS_GOOGLE_CSE_GMAIL_JWT_ISSUER", JWT_ISSUER_URI);
+    cosmian_logger::log_utils::log_init("debug,cosmian_kms_server=trace");
+
+    let jwt = generate_google_jwt().await;
+
+    let app = test_utils::test_app(Some("http://127.0.0.1/".to_string())).await;
+
+    let import_request = import_google_cse_symmetric_key();
+    tracing::debug!("import_request created");
+
+    let response: ImportResponse = test_utils::post(&app, import_request).await?;
+    tracing::debug!("import response: {response:?}");
+
+    tracing::debug!("grant post");
+    let access = Access {
+        unique_identifier: Some(UniqueIdentifier::TextString("google_cse".to_string())),
+        user_id: "*".to_string(),
+        operation_types: vec![
+            ObjectOperationType::Create,
+            ObjectOperationType::Destroy,
+            ObjectOperationType::Get,
+            ObjectOperationType::Encrypt,
+            ObjectOperationType::Decrypt,
+        ],
+    };
+
+    let access_response: SuccessResponse =
+        test_utils::post_with_uri(&app, access, "/access/grant").await?;
+    tracing::debug!("grant response post: {access_response:?}");
+
+    // The RSA blue private key has been AES256 wrapped with `demo.key.json`
+    let wrapped_private_key =
+        include_str!("../../../../../documentation/docs/google_cse/blue_wrapped_private_key");
+    let digest = "MSKaRPMiIFwZoWGYjA/MV8mLNNYGW3GpODrEjdbbQqE=";
+
+    let pksr = PrivateKeySignRequest {
+        authentication: jwt.clone(),
+        authorization: jwt,
+        algorithm: "SHA256withRSA".to_string(),
+        digest: digest.to_string(),
+        e_key: Some("e_key".to_string()),
```

Comment:
remove this

Reply:
done in a83e9638f1de6b310f771e776498c958c1d9f7d6